File: backend/validator/LoginValidator.php
<?php
// backend/validators/LoginValidator.php

require_once 'ValidatorBase.php';

class LoginValidator extends ValidatorBase {
  
  public static function validate($postData): array {
    $validator = new self($postData); // Composición con base

    $fieldsToValidate = [
        'email' => trim($postData['email'] ?? ''),
        'contrasena' => $postData['contrasena'] ?? '' 
    ];

    // Usar el método heredado de ValidatorBase
    $result = $validator->validateFields($fieldsToValidate);

    if (!$result['success']) {
        return [
            'success' => false, 
            'errors' => $result['errors']
        ];
    }

    return [
        'success' => true,
        'cleanData' => $result['cleanData']
    ];
  }
}
?>

File: backend/validator/LoginValidator.go
package validator

import "net/http"

type LoginValidator struct {
	ValidatorBase
}

var LoginValidatorInstance = &LoginValidator{
	ValidatorBase: ValidatorBase{
		Rules: map[string]Rule{
			"email": {
				Required: true,
				Email:    true,
			},
			"password": {
				Required:  true,
				MinLength: 6,
			},
		},
	},
}

func (v *LoginValidator) Validate(r *http.Request) ValidationResult {
	return v.ValidatorBase.Validate(r)
}
